### PR TITLE
Fix error when custom check bins are provided during onprem install

### PR DESCRIPTION
**Pull request by branden**
## High-level description

This fixes an error where, when custom check bins are provided, the installer will attempt to copy the custom check bins package from the Docker container's artifact cache, and fail with an error when it can't find it. This is accomplished by adding the package to the gen result's list of generated stable artifacts, which are skipped when copying artifacts. This error was introduced in bf2032fb4fdf5fc31fa0d79f860b9d0d45255a23.

This also includes a followup refactor, see commits.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-1982](https://jira.mesosphere.com/browse/DCOS_OSS-1982) genconf fails when custom check bins are provided

## Checklist for all PRs

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: Tested manually on CentOS 7.2. Ticket for getting this covered by CI here: https://jira.mesosphere.com/browse/QUALITY-1724
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Comment by branden** [comment]
@mesosphere-mergebot label ready for review

File: gen/build_deploy/util.py [hunk 1]
```diff
@@ -29,15 +29,15 @@ def copy_makedirs(src, dest):
     shutil.copy(src, dest)
 
 
-def do_bundle_onprem(extra_files, gen_out, output_dir):
+def do_bundle_onprem(gen_out, output_dir):
```
Comment (orsenthil) [inline gen/build_deploy/util.py]: Is there no coverage for this? Perhaps next time we touch this code, adding tests/ increasing coverage would be good idea.

Comment (branden) [reply gen/build_deploy/util.py]: Unit testing this function wouldn't add much value, it's a helper that doesn't encapsulate much. It'd be one of those tests that's effectively a copy of the function it's testing, but with a bunch of mocks and asserts. It wouldn't do anything to guarantee that the logic in the function is correct with regard to the rest of gen and the installer, and you'd have to tediously update it every time you changed the original code. To really test this function you have to provide it with the output of an onprem genconf run, and then execute a deployment from the resulting `serve/` dir that it creates. Basically you need an integration test.

**Comment by mesosphere-teamcity** [comment]
Corresponding DC/OS PR: https://github.com/dcos/dcos/pull/2449

**Pull request merged**